src/org/mozilla/javascript/JavaAdapter.java

```diff
@@ -501,6 +501,7 @@ private static void appendOverridableMethods(Class<?> c,
             ArrayList<Method> list, HashSet<String> skip)
     {
         Method[] methods = c.getDeclaredMethods();
+        if(c.isInterface())methods=c.getMethods();
```

Comment:
In this case, `c.getDeclaredMethods()` is unnecessary.
This would be better.
https://github.com/mozilla/rhino/blob/8442718d94434e4e1c054b6228d0d4b1603b2878/src/org/mozilla/javascript/JavaAdapter.java#L491
```java
appendOverridableMethods(c.getDeclaredMethods(), list, skip); 
```
https://github.com/mozilla/rhino/blob/8442718d94434e4e1c054b6228d0d4b1603b2878/src/org/mozilla/javascript/JavaAdapter.java#L495
```java
appendOverridableMethods(c.getMethods(), list, skip); 
```